sql/core/src/test/scala/org/apache/spark/sql/streaming/EventTimeWatermarkSuite.scala

```diff
@@ -302,6 +302,71 @@ class EventTimeWatermarkSuite extends StreamTest with BeforeAndAfter with Matche
     )
   }
 
+  test("multiple aggregates in append mode") {
+    val inputData = MemoryStream[Int]
+
+    val windowedAggregation = inputData.toDF()
+      .withColumn("inputtime", $"value".cast("timestamp"))
+      .withWatermark("inputtime", "10 seconds")
+      .groupBy(window($"inputtime", "5 seconds") as 'window1, $"inputtime").count()
+      .select($"window1.end".as("windowtime"), $"count".as("num"))
+      .withWatermark("windowtime", "5 seconds")
+      .groupBy(window($"windowtime", "5 seconds") as 'window2, $"num").count()
+      .select($"window2.start".cast("long").as[Long], $"num", $"count")
+
+    testStream(windowedAggregation)(
+      AddData(inputData, 10, 11, 11, 12, 12),
+      CheckNewAnswer(),
+      AddData(inputData, 25), // watermark -> group1 = 15, group2 = 10
+      CheckNewAnswer(),
+      assertNumTotalStateRows(3),
+      AddData(inputData, 26, 26, 27),
+      CheckNewAnswer(),
+      AddData(inputData, 40), // watermark -> group1 = 30 , group2 = 25
+      CheckNewAnswer((15, 1, 1), (15, 2, 2))
```

Comment:
This query doesn't look intuitive and looks a bit hard to track since second aggregation groups aggregated value on first aggregation.

Could we change the query a bit? I guess one of example would be having `user` as column, and grouping by `window` and `user` in the first aggregation (so aggregated per `window` and `user`), and grouping by `window` in the second aggregation (so aggregated per `window` - global users).

Reply:
I wanted to check with multiple keys as part of the second group as well. I have added comments so that its easy to follow. Also added an additional test case like the one you suggested, with required comments.